implementations/vault/VaultBaselineY.sol

```diff
@@ -0,0 +1,132 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.6.2;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/GSN/Context.sol";
+
+import "../../interfaces/flamincome/Strategy.sol";
+
+contract VaultBaselineY is ERC20 {
+    using SafeERC20 for IERC20;
+    using Address for address;
+    using SafeMath for uint256;
+
+    IERC20 public token;
+
+    uint public min = 9500;
+    uint public constant max = 10000;
+
+    address public governance;
+    address public strategy = address(0);
+    address public xvault;
+
+    constructor (address _token, address _xvault) public ERC20(
+        string(abi.encodePacked("YIELD FLAMINCOME ", ERC20(_token).name())),
+        string(abi.encodePacked("Y", ERC20(_token).symbol()))
+    ) {
+        _setupDecimals(ERC20(_token).decimals());
+        token = IERC20(_token);
+        governance = msg.sender;
+        xvault = _xvault;
+    }
+
+    function balance() public view returns (uint) {
+        return token.balanceOf(address(this))
+                .add(Strategy(strategy).balanceOf())
+                .sub(IERC20(xvault).totalSupply());
+    }
+
+    function setMin(uint _min) public {
+        require(msg.sender == governance, "!governance");
+        min = _min;
+    }
+
+    function setGovernance(address _governance) public {
+        require(msg.sender == governance, "!governance");
+        governance = _governance;
+    }
+
+    function setXVault(address _xvault) public {
+        require(msg.sender == governance, "!governance");
+        xvault = _xvault;
+    }
+
+    function setStrategy(address _strategy) public {
+        require(msg.sender == governance, "!governance");
+
+        if (strategy != address(0)) {
+           Strategy(strategy).withdrawAll();
```

Comment:
del withdrawall.
strategy add a method update(): transerfAll token to new strategy add modify X and Y's strategy